streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatter.java

```diff
@@ -266,6 +276,11 @@ private static class PartiallyDeserializedSegmentValue implements SegmentValue {
         private int deserIndex = -1; // index up through which this segment has been deserialized (inclusive)
         private List<TimestampAndValueSize> unpackedReversedTimestampAndValueSizes;
         private List<Integer> cumulativeValueSizes; // ordered same as timestamp and value sizes (reverse time-sorted)
+        private int valuesStartingIndex = -1; // the index of first value in the segment
```

Comment:
"first value" based on ts, right? but last value in the list... might be good to extend the comment